File: carDealership/src/main/java/com/carDealership/carDealership/models/Transaction.java
package com.carDealership.carDealership.models;

import com.carDealership.carDealership.enums.PaymentMethod;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Entity
@Table(name="transakcje")
@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
public class Transaction {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    int id;

    @Column(name="data_transakcji")
    LocalDateTime transactionDate;

    @Column(name="kwota")
    BigDecimal totalAmount;

    @Column(name="sposob_platnosci")
    @Enumerated(EnumType.STRING)
    PaymentMethod paymentMethod;

    @Column(name="czy_zarejestrowany")
    boolean registered;

    @Column(name="czy_ubezpieczony")
    boolean insured;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name="client_id")
    Client client;

    @OneToOne(fetch = FetchType.EAGER)
    @JoinColumn(name="car_id")
    Car car;

}

File: salon_samochodowy/lib/views/classes/transaction.dart
import 'car.dart';
import 'client.dart';

class Transaction {
  final int id;
  final DateTime transactionDate;
  final double totalAmount;
  final String paymentMethod;
  final bool registered;
  final bool insured;
  final Client client;
  final Car car;

  Transaction({
    required this.id,
    required this.transactionDate,
    required this.totalAmount,
    required this.paymentMethod,
    required this.registered,
    required this.insured,
    required this.client,
    required this.car,
  });

  factory Transaction.fromJson(Map<String, dynamic> json) {
    return Transaction(
      id: json['id'],
      transactionDate: DateTime.parse(json['transactionDate']),
      totalAmount: json['totalAmount'].toDouble(),
      paymentMethod: json['paymentMethod'],
      registered: json['registered'],
      insured: json['insured'],
      client: Client.fromIndividualJson(json['client']),
      car: Car.fromJson(json['car']),
    );
  }
}
